kyaml/setters2/set_test.go

```diff
@@ -0,0 +1,437 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package setters2
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func TestSet_Filter(t *testing.T) {
+	var tests = []struct {
+		name     string
+		setter   string
+		openapi  string
+		input    string
+		expected string
+	}{
+		{
+			name:   "set-replicas",
+			setter: "replicas",
+			openapi: `
+openAPI:
+  definitions:
+    io.k8s.cli.setters.no-match-1':
+      x-k8s-cli:
+        setter:
+          name: no-match-1
+          value: "1"
+    io.k8s.cli.setters.replicas:
+      x-k8s-cli:
+        setter:
+          name: replicas
+          value: "4"
+    io.k8s.cli.setters.no-match-2':
+      x-k8s-cli:
+        setter:
+          name: no-match-2
+          value: "2"
+ `,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas"}
```

Comment:
Should be reference just be "replicas" as rest of the part can be reconstructed. By this, user can also clearly see the setter name in config file.

Reply:
I don't think so, it wouldn't be a valid open api reference, so we would be creating our own syntax